File: src-typescript-backup/interfaceAdapters/controllers/get.controller.ts
import { Controller, Get, Inject, Query } from '@nestjs/common';
import { TodoService } from 'src/usecases/todo.service';
import { Todo } from '../../entities/todo';
import { TodoListDto } from 'src/usecases/todoList.dto';

@Controller('todo')
export class GetController {
  constructor(
    @Inject('TodoService') private readonly todoService: TodoService,
  ) {}

  @Get('list.json')
  async run(
    @Query('user_id') user_id: string,
    @Query('exclude_done_todo') exclude_done_todo?: boolean,
    @Query('fields') field?: string,
  ): Promise<TodoListDto> {
    if (!user_id) throw new Error('user_id is required');

    const fields = field ? field.split(',') : [];

    return await this.todoService.getTodoList(
      Number.parseInt(user_id),
      fields as (keyof Todo)[],
      exclude_done_todo,
    );
  }
}

File: src/main/java/com/todoapp/interfaceAdapters/controllers/GetController.java
package com.todoapp.interfaceAdapters.controllers;

import com.todoapp.usecases.dto.ITodoDto;
import com.todoapp.usecases.service.TodoService;
import com.todoapp.usecases.dto.TodoListDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/todo")
public class GetController {

    private final TodoService todoService;

    public GetController(TodoService todoService) {
        this.todoService = todoService;
    }

    @GetMapping("/list.json")
    public ResponseEntity<List<ITodoDto>> getTodoList(@RequestParam(value = "exclude_done_todo") boolean excludeDoneTodo, @RequestParam Optional<String> fields) {
        TodoListDto todoList = todoService.getTodoList(excludeDoneTodo, fields);
        return ResponseEntity.ok(todoList.todos());
    }
} 